src/raster/array.jl

```diff
@@ -0,0 +1,38 @@
+Base.size(dataset::Dataset) = begin
+    band = getband(dataset, 1)
+    width(band), height(band), nraster(dataset)
+end
+
+const AllowedXY = Union{Integer,Colon,AbstractRange}
+const AllowedBand = Union{Integer,Colon,AbstractArray}
+
+Base.getindex(dataset::Dataset, x::AllowedXY, y::AllowedXY, bands::AllowedBand) = begin
+    # Convert colons to ranges
+    x, y, bands = map(convertcolon, (x, y, bands), size(dataset))
+    buffer = Array{getdatatype(getband(dataset, 1)),3}(undef, length.((x, y, bands))...)
+    xoffset, yoffset = first(x) - 1, first(y) - 1
+    xsize, ysize = length(x), length(y)
+    indices = bandindices(bands)
+    read!(dataset, buffer, indices, xoffset, yoffset, xsize, ysize)
+    # Drop dimensions of integer indices
+    reshape(buffer, dropint(x, y, bands)...)
+end
+
+Base.getindex(dataset::Dataset, x::Int, y::Int, band::Int) = begin
```

Comment:
is there a reason the `::Int` are not `::Integer`s here?

Reply:
Base uses `Int`:
https://github.com/JuliaLang/julia/blob/master/base/array.jl#L758

But I'm not sure why.